lib/timers.js

```diff
@@ -10,36 +10,84 @@ const kOnTimeout = Timer.kOnTimeout | 0;
 // Timeout values > TIMEOUT_MAX are set to 1.
 const TIMEOUT_MAX = 2147483647; // 2^31-1
 
-// IDLE TIMEOUTS
+
+// HOW and WHY the timers implementation works the way it does.
+//
+// Since many parts of Node.js and user applications rely on timeouts, there may
+// be a very large amount of timeouts scheduled at any given time.
+// As such, it is very important that the timers implementation be very fast and
+// efficent.
```

Comment:
Super-micro-nits: `fast and efficient` is kind of redundant here. You can probably just say `fast`. Or `efficient`. You can probably also remove both instances of `very` from the sentence. Lastly, I think `therefore` is more natural and less confusing than `as such`. `Therefore, it is important that the timers implementation be efficient.`

I promise not to do this for every sentence in the comments.